### PR TITLE
Add split-video-scenes extension

### PR DESCRIPTION
**Pull request by remorses**

### CODE REVIEW

File: extensions/split-video-scenes/package.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  "icon": "command-icon.png",
+  "author": "xmorse",
+  "categories": [
+    "Productivity"
+  ],
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "Split Video Into Scenes",
```
Comment (pernielsentikaer) [inline extensions/split-video-scenes/package.json:15]:
```suggestion
      "title": "Split Video into Scenes",
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- Can we somehow improve this flow? I didn't have pip so I kinda stranded - is there a homepage we can land on instead for guiding the user?

I also tried with pip3 since it was installed but it didn't go

Raycast26082024-V4neIoKj png@2x

- I would also add a bit more to the `README` to explain what it does and which modes do what

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by remorses** [comment]
I updated the extension to install the command with pipx, so there is no longer the problem you faced. pipx creates a sandbox for each installed lib, so it does not create problem for the user python environment.

I also updated the code to detach the scenedetect process so an user can use other extensions while scenedetect is running (which can take minutes), if the user runs the extension while the process is in progress it will show the current logs. here is a demo

https://github.com/user-attachments/assets/60f3fe21-2634-4609-ad3d-a600456bd956

**Comment by pernielsentikaer** [comment]
I think we might need to let the user handle this installation process. I ended up in a loop because something was wrong with my PATHs, and I was unable to see the message since the window closed. I had to record the screen in order to read it

Raycast27082024-xDpga2sN png@2x

**Comment by remorses** [comment]
I added the .local/bin folder in path, this should fix it permanently because pipx always installs packages there

File: extensions/split-video-scenes/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+          <Action.SubmitForm onSubmit={handleSubmit} title="Split Video into Scenes" />
+        </ActionPanel>
+      }
+    >
+      <Form.FilePicker
+        id="file"
+        canChooseDirectories={false}
+        canChooseFiles={true}
+        autoFocus
+        title="Select video files"
```
Comment (pernielsentikaer) [inline extensions/split-video-scenes/src/index.tsx:255]:
```suggestion
        title="Select Video Files"
```

File: extensions/split-video-scenes/src/index.tsx [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+        shell: true,
+        stdio: ["ignore", outputLogFile, outputLogFile],
+        detached: true,
+      });
+      childProcess.unref();
+      attachToLogsFile();
+      exitCode = await new Promise((resolve) => childProcess!.on("close", (code) => resolve(code || 0)));
+    }
+    if (exitCode === 0) {
+      closeMainWindow();
```
Comment (pernielsentikaer) [inline extensions/split-video-scenes/src/index.tsx:168]: Should we popToRoot and show a toast instead(with actions to open folder) - it felt weird that the window just closed without anything

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
That worked as expected 🔥

I gave it another spin and found a few things 🔥

File: extensions/split-video-scenes/src/index.tsx (hunk 3 above)
Comment (remorses) [reply extensions/split-video-scenes/src/index.tsx]: Now it shows an HUD and the folder with the scenes in finder

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**